### PR TITLE
Update AEPAudience to major version 4.0, minimum Xcode version to 14.1 and iOS 11

### PR DESCRIPTION
**Pull request by cacheung**
Update AEPAudience to major version 4.0, minimum Xcode version to 14.1 and iOS 11

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -20,3 +20,10 @@ disabled_rules: # rule identifiers to exclude from running
   - type_name
   - large_tuple
   - trailing_comma
+
+analyzer_rules:
+- unused_import
+
+type_body_length:
+    warning: 50
+    error: 400
```
Comment (emdobrin) [inline .swiftlint.yml:29]: this should be max 150

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,42 +3,46 @@
...
+  install_dependencies:
     steps:
-      - checkout
-
       # restore pods related caches
       - restore_cache:
+          name: Restoring Gemfile Cache
           keys:
-            - 2-gems-{{ checksum "Gemfile.lock" }}
+            - 1-gems-{{ checksum "Gemfile.lock" }}
```
Comment (emdobrin) [inline .circleci/config.yml:29]: shouldn't this be incremented to 3-*

File: .swiftlint.yml [hunk 3]
```diff
@@ -2,7 +2,6 @@ excluded: # paths to ignore during linting. Takes precedence over `included`.
   - AEPAudience/Tests
```
Comment (kevinlind) [inline .swiftlint.yml:2]: SwiftLint should run on the Tests as well, so this line can be removed and the Xcode project setup to run SwiftLint the the build phase of the tests. We can do this in another PR if desired as the lint changes may touch a lot of files.

**Review by emdobrin — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -8,60 +8,66 @@ SIMULATOR_ARCHIVE_DSYM_PATH = $(CURR_DIR)/build/ios_simulator.xcarchive/dSYMs/
...
 test-SPM-integration:
-	(sh ./Script/test-SPM.sh)
+	sh ./Script/test-SPM.sh
 
 test-podspec:
-	(sh ./Script/test-podspec.sh)
-
-pod-lint:
-	(pod lib lint --allow-warnings --verbose --swift-version=5.1)
+	sh ./Script/test-podspec.sh
```
Comment (emdobrin) [inline Makefile:73]: nit: can we add a step for bump version 
```
# make bump-versions from='3\.1\.0' to=3.1.1
bump-versions:
	(LC_ALL=C find . -type f -name 'project.pbxproj' -exec sed -i '' 's/$(from)/$(to)/' {} +)
	(LC_ALL=C find . -type f -name '*.swift' -exec sed -i '' 's/$(from)/$(to)/' {} +)
	(LC_ALL=C find . -type f -name '*.podspec' -exec sed -i '' 's/$(from)/$(to)/' {} +)
```

**Review by emdobrin — APPROVED** [review]
